File: client/src/lib/invoke/sendTransaction.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { ClassicTransactionSent, Config } from "@bindings";

export function sendTransaction(
    server_pubkey: string,
    config: Config,
    transaction: ClassicTransactionSent
): ResultAsync<void, string> {
    return ResultAsync.fromPromise(
        invoke("send_transaction", { server_pubkey, config, transaction }),
        (error) => `Failed to send transaction: ${error}`
    );
}

File: client/src-tauri/src/commands/send_transaction.rs
use crate::{
    config::Config, nexium_api::send_transaction as send_transaction_api,
    nexium_api::ClassicTransactionSent,
};

#[tauri::command]
pub async fn send_transaction(
    server_pubkey: String,
    config: Config,
    transaction: ClassicTransactionSent,
) -> Result<(), String> {
    tauri::async_runtime::spawn_blocking(move || {
        send_transaction_api(server_pubkey, transaction, config)
    })
    .await
    .map_err(|err| format!("Failed to send transaction: {}", err))?
}
